pkg/oomstore/export.go

```diff
@@ -139,3 +139,69 @@ func (s *OomStore) ChannelExportStream(ctx context.Context, opt types.ChannelExp
 	header := append([]string{group.Entity.Name}, features.Names()...)
 	return types.NewExportResult(header, stream, exportErr), nil
 }
+
+// ChannelExport exports the latest streaming feature values up to the given timestamp.
+// Currently, this API can only export features in one feature group.
+func (s *OomStore) ChannelExport(ctx context.Context, opt types.ChannelExportOpt) (*types.ExportResult, error) {
+	if err := validateFeatureFullNames(opt.FeatureFullNames); err != nil {
+		return nil, errdefs.WithStack(err)
+	}
+	features, err := s.ListFeature(ctx, types.ListFeatureOpt{
+		FeatureFullNames: &opt.FeatureFullNames,
+	})
+	if err != nil {
+		return nil, errdefs.WithStack(err)
+	}
+
+	featureMap := buildGroupIDToFeaturesMap(features)
+	snapshotTables := make(map[int]string)
+	cdcTables := make(map[int]string)
+	for _, featureList := range featureMap {
+		group := featureList[0].Group
+		revisions, err := s.ListRevision(ctx, &group.ID)
+		if err != nil {
+			return nil, errdefs.WithStack(err)
```

Comment:
```suggestion
			return nil, err
```

Reply:
Because this is not where the error occurred

Reply:
why not wrap it with `errdefs.WithStack`?

Reply:
This is an error in `s.ListRevision`, where `errdefs.WithStack` has already been executed internally

Reply:
gotcha!